Uplift/Controllers/GymDetailViewController.swift

```diff
@@ -282,6 +287,31 @@ func getHoursHeight() -> CGFloat {
     }
 }
 
+extension GymDetailViewController {
+    func categorizeEquipment(equipmentList: [Equipment]) -> [EquipmentCategory] {
+        var equipmentDictionary: [String: [Equipment]] = [:]
+        let sorted = equipmentList.sorted(by: { $0.equipmentType < $1.equipmentType })
+        var category = sorted[0].equipmentType
+        equipmentDictionary[category] = []
+
+        sorted.forEach { (equipment) in
+            if equipment.equipmentType == category {
+                equipmentDictionary[category]?.append(equipment)
+            } else {
+                category = equipment.equipmentType
+                equipmentDictionary[category] = [equipment]
+            }
+        }
```

Comment:
the logic here seems to be more complicated than it has to be. essentially we want to build `equipmentDictionary` to be a map from a category to all equipments in that category. the current logic sorts the `equipmentList` by `equipmentType` and then we maintain the current `category` to build this dictionary. instead, we can simply just do:
```
let equipmentDictionary = equipmentList.reduce([:], { dict, equipment in 
  if equipment.equipmentType in dict {
    dict[equipment.equipmentType]?.append(equipment)
  } else {
    dict[equipment.equipmentType] = [equipment]
  }
}
```